app/uk/gov/hmrc/api/testlogin/controllers/LoginController.scala

```diff
@@ -42,20 +46,29 @@ class LoginController @Inject()(override val messagesApi: MessagesApi, loginServ
   )
 
   def showLoginPage(continue: String) = Action.async { implicit request =>
-    successful(Ok(uk.gov.hmrc.api.testlogin.views.html.login(continue)))
+    if(continueUrlService.isValidContinueUrl(continue)) {
+      successful(Ok(login_template(continue)))
+    } else {
+      successful(BadRequest(error_template("", "", "Invalid Parameters")))
```

Comment:
Could this be shared - see below

Reply:
I thought about that too but couldn't work out what the right combination of type parameterised implicits that I needed were!